### ISSUE TITLE
DeltaTable class name collision

### ISSUE THREAD

**Issue by MrPowers** [delta-io/delta-rs#438]
# Description

We're working on building a Dask Delta Lake reader.  I was writing an integration test that wrote out a Delta Lake with [delta](https://github.com/delta-io/delta) and read it in with [delta-rs](https://github.com/delta-io/delta-rs).  The test was behaving unexpectedly because `DeltaTable` is defined in both repos. 

I don't think these libraries will normally be used in conjunction, but think it's wise to avoid name collisions in any case.  Also think different name will make StackOverflow questions easier and debugging with Googling easier.

Thanks for making a great lib.

**Comment by houqp** [comment]
Thanks @MrPowers for the report. The concern of SEO is certainly valid. I also don't have a good solution in mind. You are right that they aren't intended to be used in conjunction. For the rare case that they need to be used together, we will simply have to rely on python package namespace or import one of the DeltaTable with alias.

Another way to look at this could be viewing `DeltaTable` as a generic search term like `Dataframe` and let users search `DeltaTable` together with project name. For example `pyspark DeltaTable` or `native DeltaTable`. There is also another [pure python deltalake implementation](https://pypi.org/project/delta-lake-reader/) that also uses `DeltaTable` as the class name, so even if we change to something else, people will still get confused.

**Comment by MrPowers** [comment]
@houqp - I'm building a [dask-interop](https://github.com/MrPowers/dask-interop) project that writes out Delta files with Spark and reads them in with delta-rs.  I just created different conda environments as a work-around.  Not ideal, but like you mentioned, not a common issue, so think it's fine as-is.

On a happy note, I was able to use the delta-rs API to easily read a Delta table into a Dask DataFrame, great work building this lib!
